contracts/adapters/OptimisticAuctionRebalanceExtensionV1.sol

```diff
@@ -0,0 +1,406 @@
+     */
+    function startRebalance(
+        IERC20 _quoteAsset,
+        address[] memory _oldComponents,
+        address[] memory _newComponents,
+        AuctionExecutionParams[] memory _newComponentsAuctionParams,
+        AuctionExecutionParams[] memory _oldComponentsAuctionParams,
+        bool _shouldLockSetToken,
+        uint256 _rebalanceDuration,
+        uint256 _positionMultiplier
+    )
+        external
+        override
+        onlyIfOpen()
+    {
+        bytes32 proposalHash = keccak256(abi.encode(
+            setToken,
+            _quoteAsset,
+            _oldComponents,
+            _newComponents,
+            _newComponentsAuctionParams,
+            _oldComponentsAuctionParams,
+            _shouldLockSetToken,
+            _rebalanceDuration,
+            _positionMultiplier
+        ));
+
+        bytes32 assertionId = assertionIds[proposalHash];
+        // Disputed assertions are expected to revert here. Assumption past this point is that there was a valid assertion.
+        require(assertionId != bytes32(0), "Proposal hash does not exist");
+
+        _deleteProposal(assertionId);
+        
+        // There is no need to check the assertion result as this point can be reached only for non-disputed assertions.
+        // It is expected that future versions of the Optimistic Oracle will always revert here, 
+        // if the assertionId has not been settled and can not currently be settled.
+        productSettings.optimisticParams.optimisticOracleV3.settleAndGetAssertionResult(assertionId);
+
+        address[] memory currentComponents = setToken.getComponents();
+        
+        require(currentComponents.length == _oldComponents.length, "Mismatch: old and current components length");
+
+        for (uint256 i = 0; i < _oldComponents.length; i++) {
+            require(currentComponents[i] == _oldComponents[i], "Mismatch: old and current components");
+        }
+
+        bytes memory callData = abi.encodeWithSelector(
+            IAuctionRebalanceModuleV1.startRebalance.selector,
+            setToken,
+            _quoteAsset,
+            _newComponents,
+            _newComponentsAuctionParams,
+            _oldComponentsAuctionParams,
+            _shouldLockSetToken,
+            _rebalanceDuration,
+            _positionMultiplier
+        );
+
+        invokeManager(address(auctionModule), callData);
+        isOpen = false;
+    }
+
+     // Constructs the claim that will be asserted at the Optimistic Oracle V3.
+    function _constructClaim(bytes32 proposalHash, bytes32 rulesHash) internal pure returns (bytes memory) {
+        return
+            abi.encodePacked(
+                AncillaryData.appendKeyValueBytes32("", PROPOSAL_HASH_KEY, proposalHash),
+                ",",
+                RULES_KEY,
+                ":\"",
+                rulesHash,
+                "\""
+            );
+    }
```

Comment:
can we add some comments here? maybe link to UMA resource on claim construction?

Reply:
I couldn't really find any specific rules on claim construction on the uma documentation.
As far as I understand we can define our own conventions on the formatting / interpretation of claims.
The UMA examples from the docs are usually english sentences such as:
- "The sky is blue"
- "Argentinia won the 2022 Fifa World Cup"
- "The price of X Token was Y USD at time Z"

Ideally we should have a resource of our own where this formatting will be defined. (Or maybe even a frontend app in the future which shows the correct interpretation of those parameters).

Reply:
Added a comment with a link tot he OptimisticGovernor implementation: https://github.com/IndexCoop/index-coop-smart-contracts/pull/156/commits/ced1ad5a55b5b2eb00b8642028d9ab9ce3fd25c0